test/jdk/java/awt/print/PrinterJob/PrinterDevice.java

```diff
@@ -54,39 +66,52 @@ public static void main(String args[]) throws PrinterException {
         aset.add(OrientationRequested.LANDSCAPE);
         pj.setPrintable(new PrinterDevice());
         pj.print(aset);
+        if (failed) {
+            throw new RuntimeException("Test failed but no exception propagated.");
+        }
```

Comment:
A comment that `pj.print` should not throw exception would suffice, even though it's implied by jtreg any way.

This statement is essentially unreachable if `failed` is set to `true`.

Reply:
On JDK 17 and earlier you will reach this line with this test as written
And without that exception, the test would pass when it should fail.

Reply:
With JDK 17, I still see the exception:

```text
Exception in thread "main" java.awt.print.PrinterException
	at java.desktop/sun.lwawt.macosx.CPrinterJob.print(CPrinterJob.java:393)
	at PrinterDevice.main(PrinterDevice.java:68)
Caused by: java.lang.NullPointerException: Cannot read field "m00" because "Tx" is null
	at java.desktop/java.awt.geom.AffineTransform.<init>(AffineTransform.java:490)
	at java.desktop/sun.print.PrinterGraphicsConfig.getDefaultTransform(PrinterGraphicsConfig.java:105)
	at PrinterDevice.print(PrinterDevice.java:87)
```

Yet it's fine to ensure no exception is swallowed.

Reply:
That exception you are seeing is from the printStackTrace()
If you were to comment out these lines as shown below
@@ -90,9 +90,9 @@ public int print(Graphics g, PageFormat pf, int pageIndex) {
             AffineTransform gt = g2.getTransform();
             System.out.println("Graphics2D transform = " + gt);
         } catch (Exception e) {
-            failed = true;
-            System.err.println("Unexpected exception getting transform.");
-            e.printStackTrace();
+            //failed = true;
+            //System.err.println("Unexpected exception getting transform.");
+            //e.printStackTrace();
             throw e;

then the test would pass on JDK 17

Reply:
Nope, I checked that it's the real exception. Yet I used a recent version of JDK 17 where [JDK-8262731](https://bugs.openjdk.org/browse/JDK-8262731) was fixed, it's present in 17.0.1.

Indeed, the test passes with GA version of JDK 17 without the `failed` field.